### PR TITLE
make time components polling components

### PR DESCRIPTION
**Pull request by badbadc0ffee**
## Description:

I'd suggest that real time components should be implemented as polling components so that the `update_interval` can be configured in the same way as for other components (where applicable, the sntp will run in the `loop()` instead of `update()` e.g.).

## Checklist:
  - [X] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs): esphome/esphome-docs#924

### CODE REVIEW

**Comment by badbadc0ffee** [comment]
this is a follow-up to #1441

**Comment by glmnet** [comment]
why?

File: esphome/components/gps/time/__init__.py [hunk 1]
```diff
@@ -6,12 +6,12 @@
...
 DEPENDENCIES = ['gps']
 
-GPSTime = gps_ns.class_('GPSTime', time_.RealTimeClock, GPSListener)
+GPSTime = gps_ns.class_('GPSTime', cg.PollingComponent, time_.RealTimeClock, GPSListener)
 
 CONFIG_SCHEMA = time_.TIME_SCHEMA.extend({
     cv.GenerateID(): cv.declare_id(GPSTime),
     cv.GenerateID(CONF_GPS_ID): cv.use_id(GPS),
-}).extend(cv.COMPONENT_SCHEMA)
+}).extend(cv.polling_component_schema('5min'))
```
Comment (glmnet) [inline esphome/components/gps/time/__init__.py:14]: isn't TIME_SCHEMA extending polling_component_schema already?

Comment (badbadc0ffee) [reply esphome/components/gps/time/__init__.py:14]: It is. But the default update interval for the GPS time component previously was 5 minutes while the other components have a default of 15 minuted. I didn't want to change that default behavior.

Comment (badbadc0ffee) [reply esphome/components/gps/time/__init__.py:14]: It is. But the default update interval for the GPS time component previously was 5 minutes while the other components have a default of 15 minuted. I didn't want to change that default behavior.

Comment (glmnet) [reply esphome/components/gps/time/__init__.py:14]: so this just overrides that, no side effects?
Never used the validator like this.

Comment (badbadc0ffee) [reply esphome/components/gps/time/__init__.py:14]: It does exactly what it is intended to do. ;-)

Comment (glmnet) [reply esphome/components/gps/time/__init__.py:14]: ok then, I've seen this differently before: you have to export TIME_SCHEMA as a function which receives args and then you pass down the args to the extended components, etc.

But if this works as intended not problem

**Pull request merged**